src/realm/group.cpp

```diff
@@ -141,11 +141,27 @@ Group::Group(SlabAlloc* alloc) noexcept
     init_array_parents();
 }
 
+class TableRecycler : public std::vector<Table*> {
+public:
+    ~TableRecycler()
+    {
+        for (auto t : *this) {
```

Comment:
Could it be argued that we should rather have a REALM_UNREACHABLE() here?

Reply:
Makes sense.